### DISCUSSION TITLE
Manage status by roles

### DISCUSSION THREAD

**Discussion by Causeofhell** [filamentphp/filament#12864] [Help] [answered]
### Package

Table builder

### Package Version

v3.2

### How can we help you?

I want to manage model status, some status can be modified only by certain role users, i have shield plugin installed and i have the custom permissions created.
how can i check if the user has permissions and then show a modal? i have tried with aprovals plugin but is not working so i ended up trying by created some actions like this one
`Action::make('approve')
                ->label('Approve')
                ->icon('heroicon-s-check-circle')
                ->color('success')
                ->action(function ($record) {
                    $record->update(['status' => LoanStatus::Approved->value]);
                    $record->save();
                })
                ->requiresConfirmation()
                ->modalHeading('Approve Loan')
                ->modalDescription('Are you sure you want to approve this loan?'),`

**Comment by leandrocfe** [comment]
If you want to show/hide an action by role

```php
->visible(auth()->user()->hasRole('super_admin'))
```

**Reply by Causeofhell** [reply]
here is the problem i am facing,
if do this i can acces to the trait declarations like can() or hasRoles()
$user = User::find($record->user_id);
$user->hasRole('super_admin');

but if i do that auth()->user()->hasRole('super_admin'); gives me an error

i have checked if i am using the same instance and also cleared the cache but the error continues....

**Reply by Causeofhell** [reply, marked as answer]
i found this is an error of intelephense Undefined method 'hasRole'.intelephense(P1013)  after disable the plugin installing another one and then restarting the vscode all goes okay
